pkg/controller/jobs/rayjob/rayjob_webhook.go

```diff
@@ -0,0 +1,131 @@
+
+var _ webhook.CustomDefaulter = &RayJobWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *RayJobWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*rayjobapi.RayJob)
+	log := ctrl.LoggerFrom(ctx).WithName("rayjob-webhook")
+	log.V(5).Info("Applying defaults", "job", klog.KObj(job))
+	jobframework.ApplyDefaultForSuspend((*RayJob)(job), w.manageJobsWithoutQueueName)
+	return nil
+}
+
+// +kubebuilder:webhook:path=/validate-ray-io-v1alpha1-rayjob,mutating=false,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayjobs,verbs=update,versions=v1alpha1,name=vrayjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &RayJobWebhook{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *RayJobWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*rayjobapi.RayJob)
+	log := ctrl.LoggerFrom(ctx).WithName("rayjob-webhook")
+	log.Info("Validating create", "job", klog.KObj(job))
+	return w.validateCreate(job).ToAggregate()
+}
+
+func (w *RayJobWebhook) validateCreate(job *rayjobapi.RayJob) field.ErrorList {
+	var allErrors field.ErrorList
+	kueueJob := (*RayJob)(job)
+
+	if w.manageJobsWithoutQueueName || jobframework.QueueName(kueueJob) != "" {
+		spec := &job.Spec
+		specPath := field.NewPath("spec")
+
+		// Should always delete the cluster after the sob has ended, otherwise it will continue to the queue's resources.
+		if !spec.ShutdownAfterJobFinishes {
+			allErrors = append(allErrors, field.Invalid(specPath.Child("shutdownAfterJobFinishes"), spec.ShutdownAfterJobFinishes, "a kueue managed job should delete the cluster after finishing"))
+		}
+
+		// Should not want existing cluster. Keuue (workload) should be able to control the admission of the actual work, not only the trigger.
+		if len(spec.ClusterSelector) > 0 {
+			allErrors = append(allErrors, field.Invalid(specPath.Child("clusterSelector"), spec.ClusterSelector, "a kueue managed job should not use an existing cluster"))
+		}
+
+		clusterSpec := spec.RayClusterSpec
+		clusterSpecPath := specPath.Child("rayClusterSpec")
+
+		// Should not use auto scaler. Once the resources are reserved by queue the cluster should do it's best to use them.
+		if pointer.BoolDeref(clusterSpec.EnableInTreeAutoscaling, false) {
+			allErrors = append(allErrors, field.Invalid(clusterSpecPath.Child("enableInTreeAutoscaling"), clusterSpec.EnableInTreeAutoscaling, "a kueue managed job should not use autoscaling"))
+		}
+
+		// Should limit the worker count to 8 - 1 (max podSets num - cluster head)
+		if len(clusterSpec.WorkerGroupSpecs) > 7 {
+			allErrors = append(allErrors, field.TooMany(clusterSpecPath.Child("workerGroupSpecs"), len(clusterSpec.WorkerGroupSpecs), 7))
```

Comment:
Does WorkerGroupSpecs have a limit? We could consider bumping the limit of Workload to match it.

Reply:
no, no limit on the ray side

Reply:
can you open an issue on their side? That's not a good API practice.